src/net_processing.cpp

```diff
@@ -1642,6 +1644,21 @@ void PeerManagerImpl::FinalizeNode(const CNode& node)
     LogPrint(BCLog::NET, "Cleared nodestate for peer=%d\n", nodeid);
 }
 
+bool PeerManagerImpl::HasDesirableServiceFlags(ServiceFlags services)
```

Comment:
Why rename from `HasAllDesirableServiceFlags`? The `HasAll` is more descriptive. It is now ambiguous whether this returns `true` if the `ServiceFlags` has some number of desirable service flags or strictly all of them.

Reply:
> Why rename from `HasAllDesirableServiceFlags`? The `HasAll` is more descriptive. It is now ambiguous whether this returns `true` if the `ServiceFlags` has some number of desirable service flags or strictly all of them.

I don't think the name is good for the PeerManager-Connman interface. What "HasAll" means for `NetEventsInterface`? There, it should be something like `bool CanConnectTo(services)` or something similar.

Still, I'm not really happy with the interface at this point. No problem on renaming it back.